lib/systems/include/controllers/DrivebrainController.h

```diff
@@ -0,0 +1,88 @@
+#ifndef __DRIVEBRAINCONTROLLER_H__
+#define __DRIVEBRAINCONTROLLER_H__
+
+#include "controllers/SimpleController.h"
+#include "SharedFirmwareTypes.h"
+#include <cmath>
+
+// TODO - [ ] need to validate that the times that are apparent in the drivebrain data
+//            and ensure that they are within tolerence to current sys-tick
+
+// TODO - [ ] if the drivebrain controller is currently the active controller,
+//            the latency fault should be latching
+
+//              meaning that if we fail any of the latency checks while active we cannot clear the timing failure fault
+//              unless we switch to another controller and back again. in reality, the CAN line or ethernet conditions
+//              probably wont improve randomly during runtime so it will keep faulting. primarily this is just to make sure
+//              we dont keep going from latent to not latent while driving and thus the driver gets jittered and the
+//              drivetrain will keep "powering up" and "powering down"
+
+// TODO - [ ] correctly measure latency between drivebrain and VCR:
+//
+
+/*! \brief Drivebrain Controller for fail-safe pass-through control of the car
+           RESPONSIBILITIES AND DOCUMENTATION
+  
+    this class is the "controller" that allows for pass-through control as commanded by the drivebrain. It also calculates the 
+    latency of the most recent input and checks to see if the most recent input is still valid and has not expired. If the 
+    input has expired then it switches over to the fail-safe control mode (MODE0) to allow for safe failing even while the car
+    is driving so that we dont lose hard-braking capabilities.
+
+    This class can clear it's own fault by switching off of this operating mode and then swapping back to this operating mode. 
+    The fault clears the first time this controller gets evaluated while switch from the swapped-to mode back to this pass 
+    through mode. 
+
+    latency measurement:
+        - the latency is measured by the difference in times in received messages from drivebrain over CAN. if the difference 
+          is too great, we swap to MODE0 control. it is assumed that the transmission delay is negligible
+
+    config: 
+        - maximum allowed latency 
+        - assigned controller mode (currently defaults to MODE4)
```

Comment:
I'd clarify that this is the assigned controller mode of *this* controller (i.e, if driver chooses MODE4, then they are in DB control mode)

Reply:
nit^